altair/utils/_vegafusion_data.py

```diff
@@ -36,21 +45,41 @@ class _ToVegaFusionReturnUrlDict(TypedDict):
     url: str
 
 
-@curried.curry
+_VegaFusionReturnType = Union[_ToVegaFusionReturnUrlDict, ToValuesReturnType]
+
+
+@overload
+def vegafusion_data_transformer(
+    data: None = ..., max_rows: int = ...
+) -> Callable[..., Any]: ...
+
+
+@overload
+def vegafusion_data_transformer(
+    data: DataFrameLike, max_rows: int
+) -> ToValuesReturnType: ...
+
+
+@overload
 def vegafusion_data_transformer(
-    data: DataType, max_rows: int = 100000
-) -> Union[_ToVegaFusionReturnUrlDict, ToValuesReturnType]:
+    data: Union[dict, pd.DataFrame, SupportsGeoInterface], max_rows: int
+) -> _VegaFusionReturnType: ...
+
+
+def vegafusion_data_transformer(
+    data: Optional[DataType] = None, max_rows: int = 100000
+) -> Union[Callable[..., Any], _VegaFusionReturnType]:
     """VegaFusion Data Transformer"""
-    if hasattr(data, "__geo_interface__"):
```

Comment:
I just noticed this won't go well. Moving this check here will break vegafusion, as it will try to parse a GeoDataFrame as a normal DataFrame, which it shouldn't do.

Reply:
@mattijn would this be better?:

```py
def vegafusion_data_transformer(
    data: Optional[DataType] = None, max_rows: int = 100000
) -> Union[Callable[..., Any], _VegaFusionReturnType]:
    """VegaFusion Data Transformer"""
    if data is None:
        return vegafusion_data_transformer
    elif isinstance(data, DataFrameLike) and not isinstance(data, SupportsGeoInterface):
        table_name = f"table_{uuid.uuid4()}".replace("-", "_")
        extracted_inline_tables[table_name] = data
        return {"url": VEGAFUSION_PREFIX + table_name}
    else:
        # Use default transformer for geo interface objects
        # # (e.g. a geopandas GeoDataFrame)
        # Or if we don't recognize data type
        return default_data_transformer(data)
```

Reply:
Yes! I think that will do.